app/src/main/java/com/example/lexis/adapters/ArticlesAdapter.java

```diff
@@ -74,18 +78,53 @@ public void bind(Article article) {
         }
 
         /*
-        Show ArticleFragment when an item is clicked.
+        Translate article if necessary and show ArticleFragment when an item is clicked.
         */
         @Override
         public void onClick(View v) {
             int position = getAdapterPosition();
-            final FragmentManager fragmentManager = fragment.getActivity().getSupportFragmentManager();
-            final Fragment articleFragment = ArticleFragment.newInstance(articles.get(position));
+            TranslateTask task = new TranslateTask();
+            task.execute(position);
+        }
+
+        // TODO: AsyncTask is deprecated so I am unsure if I should be using something else instead
```

Comment:
Yup, please use the non-deprecated [alternatives](https://stackoverflow.com/questions/58767733/android-asynctask-api-deprecating-in-android-11-what-are-the-alternatives) instead.